packages/ui-library/src/components/radio-group/index.ts

```diff
@@ -6,38 +6,122 @@ import { SizelessIconType } from '@boiler/icons';
 import { BlrFormCaptionGroupRenderFunction } from '../form-caption-group/renderFunction';
 import { BlrFormCaptionRenderFunction } from '../form-caption/renderFunction';
-import { BlrFormLabelInlineRenderFunction } from '../form-label/form-label-inline/renderFunction';
 import { TAG_NAME } from './renderFunction';
+import { TAG_NAME as RADIO_TAG_NAME } from '../radio/renderFunction';
 import { LitElementCustom } from '../../utils/lit-element-custom';
+import {
+  BlrBlurEvent,
+  BlrFocusEvent,
+  BlrRadioGroupValueChangeEvent,
+  createBlrRadioGroupValueChangeEvent,
+} from '../../globals/events';
+
+export type BlrRadioGroupEventHandlers = {
+  blrFocus?: (event: BlrFocusEvent) => void;
+  blrBlur?: (event: BlrBlurEvent) => void;
+  blrRadioGroupValueChange?: (event: BlrRadioGroupValueChangeEvent) => void;
+};
+
+/**
+ * @fires blrFocus Radio received focus
+ * @fires blrBlur Radio lost focus
+ * @fires blrRadioGroupValueChange Radio selected value changed
+ */
 
 export class BlrRadioGroup extends LitElementCustom {
   static styles = [staticFormStyles, staticRadioStyles, componentSpecificStaticStyles];
 
   @property() disabled?: boolean;
   @property() readonly?: boolean;
-  @property() checked?: boolean;
   @property() name?: string;
   @property() sizeVariant: InputSizesType = 'md';
   @property() hasLegend?: boolean;
   @property() required?: boolean;
-  @property() blrChange?: HTMLElement['oninput'];
-  @property() blrBlur?: HTMLElement['blur'];
-  @property() blrFocus?: HTMLElement['focus'];
   @property() hasError?: boolean;
   @property() errorIcon?: SizelessIconType;
-  @property() options!: RadioOption[];
   @property() hasHint = true;
   @property() groupHintMessageIcon?: SizelessIconType;
   @property() groupErrorMessage?: string;
   @property() groupHintMessage?: string;
   @property() groupErrorMessageIcon?: SizelessIconType;
   @property() legend?: string;
   @property() direction: RadioGroupDirection = 'horizontal';
-
   @property() theme: ThemeType = 'Light';
+  protected _radioElements: Element[] | undefined;
+
+  protected handleRadioClick = (event: Event) => {
+    this._radioElements?.forEach((item) => {
+      const label = item.getAttribute('label');
+      const radioInputElement = item.shadowRoot?.querySelector('input') as HTMLInputElement;
+
+      if (label === (event?.target as HTMLInputElement).id) {
+        radioInputElement.checked = true;
+      }
+    });
+
+    event.preventDefault();
+  };
+
+  firstUpdated() {
+    const slot = this.renderRoot?.querySelector('slot');
+    this._radioElements = slot?.assignedElements({ flatten: false });
+
+    setTimeout(() => {
+      {
+        this._radioElements?.forEach((item, index, elements) => {
+          if (item.localName !== RADIO_TAG_NAME) {
+            throw new Error('child component of blr-radio-group must be blr-radio');
+          }
+
+          const radioElement = <HTMLInputElement>item.shadowRoot?.querySelector('input');
```

Comment:
I agree, however, in this case completely refraining is not possible because of the styling in case of error or size-variant. Nonetheless, with the second adjustment we now need to access the shadow-dom minimum i.e. in only one place during the component initialization which I would say it pretty ok.